### ISSUE TITLE
Exception when calling first time

### ISSUE THREAD

**Issue by actionless** [google/vim-codefmt#6]
steps to reproduce:
1) open file in a suitable format
2) call `:FormatCode` or `:FormatLines` on selection

it throws:
```
ERROR(NotFound): Library "codefmtlib" could not be installed, because there are no library installers registered with maktaba.
```

3) call it again

it works as expected

**Comment by fowles** [comment]
Can you put a pointer to your vimrc somewhere?

**Comment by actionless** [comment]
https://github.com/actionless/dotfiles/blob/devel/vim/.vimrc#L73
https://github.com/actionless/dotfiles/blob/devel/vim/.vim/bundle.vim#L51

**Comment by fowles** [comment]
Awesome, will try and repro later tonight

**Comment by dbarnett** [comment]
Probably related to a maktaba bug similar to google/vim-maktaba#81. I bet you can work around it for now by adding `call maktaba#plugin#Detect()` below where you install codefmt.

**Comment by actionless** [comment]
@dbarnett 
that workaround helped me but i noticed what vim starting slower with it, so i did some "benchmarking":
http://pastebin.com/fyL9bkH5

**Comment by dbarnett** [comment]
Naturally. The reason it's been hard to solve in maktaba is that it's a challenge to solve it without slowing down vim startup. You could try `call maktaba#plugin#Get('codefmtlib')` to see if that's any less overhead, but I'm not confident it would be.

There are already a few related maktaba bugs and the whole system needs to be made more robust, but if you want to file a bug specifically for the fact that `maktaba#library#Require` fails for plugins on runtimepath, it wouldn't hurt to have the specifics captured. There might be a quick workaround for that symptom that isn't such a performance problem.

**Comment by actionless** [comment]
i think i have the same problem as in the issue you are referencing too -- i am using `stow` for my dotfiles boostrapping so my vim config is symlinked

**Comment by dbarnett** [comment]
Actually, what's the directory name of the codefmtlib plugin on your filesystem? Looking through the maktaba code, I think it works if the directory is codefmtlib/ but not anything else like vim-codefmtlib/. This issue probably doesn't have anything to do with your symlinks.

**Comment by dbarnett** [comment]
I filed google/vim-maktaba#112 specifically for this issue. I think it has to do with the directory names, not the symlink.

**Comment by actionless** [comment]
yup, directory name is `vim-codefmtlib`, Vundle uses github repo name for the folder

**Comment by dbarnett** [comment]
I have a fix in google/maktaba#113 if you want to verify that branch of maktaba solves your problem (after removing the workaround). Also would like to confirm your vim startup time is back to normal.

**Comment by actionless** [comment]
thanks, that solves my problem

startup is ok too
